airbyte-integrations/connectors/source-faker/unit_tests/unit_test.py

```diff
@@ -0,0 +1,91 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+# import jsonschema
+from airbyte_cdk.models import AirbyteMessage, Type
+from source_faker import SourceFaker
+
+
+# TODO: airbyte/airbyte-integrations/connectors/source-faker/.venv/lib/python3.9/site-packages/airbyte_cdk/sources/utils/transform.py:12: DeprecationWarning: Call to deprecated class AirbyteLogger. (Use logging.getLogger('airbyte') instead) -- Deprecated since version 0.1.47.
+# TODO: We don't have dict-with-indifferent-access in Python, so we need to make a better catalog object.  This is probably a bad idea.
+class LazyConfigDict(dict):
+    def __getattr__(self, attr):
+        return self[attr]
+
+    def __setattr__(self, attr, value):
+        self[attr] = value
```

Comment:
🐍 question:

We don't have "dict-with-indifferent-access" in Python, so we need to make a better catalog object.  This is probably a bad idea.  I'm using this below to build up mock config objects that the CDK can access with `.` notation rather than `[]` notation

Reply:
What does "dict-with-indifferent-access" mean / why do you prefer using `.` over `[]`?

Reply:
Ah! That might be a ruby-ism ((hash with indifferent access)[https://api.rubyonrails.org/v4.2/classes/ActiveSupport/HashWithIndifferentAccess.html]) - it means the object's properties can be accessed with . or [""] notation either way.

```
object.name === object['name']
```

It appeared to me that the way that the CDK loads config json files it produces an object that responds to `.` access, not `['']` access.  Without this, I get errors like `AttributeError: 'dict' object has no attribute 'streams'`.  

There might be a better constructor to make a config object that I don't know about... perhaps a parsedJson class or something like that?

Reply:
1. I think catalog should be a `ConfiguredAirbyteCatalog`, which does have the `streams` field
2. This is the right approach to implement a dict with indifferent access. Main thing to point out is that there's a few more methods that would have to be implemented for it to be fully generic (eg __del__)
3. Alternatively to (2), this stackoverflow answer shares a pretty nice way to define a dict with indifferent access without having to overload a bunch of methods https://stackoverflow.com/a/14620633

Reply:
That stackOverflow article is great, although it would need to be made recursive to work in this case with sub-properties.  Going with the real `ConfiguredAirbyteCatalog` class, seemed like the better choice (34386d38d)